python/demo/cli-mate.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import argparse
+import signal
+import sys
+from dotenv import load_dotenv
+import openai
+import os
+from aiconfig import AIConfigRuntime
+from aiconfig.model_parser import InferenceOptions
+from aiconfig.schema import Prompt
+
+
+async def run_query(question: str) -> int:
+    answer = await query(question)
+    print(answer)
+    return 0
+
+
+async def query(question: str):
+    config_file_path = "cli-mate.aiconfig.json"
+    runtime = AIConfigRuntime.load(config_file_path)
+    resolved = await runtime.resolve("question", {"prompt": question})
+    # print(f"resolved:\n{resolved}")
+
+    inference_options = InferenceOptions(stream=True)
+
+    params = {
+        "prompt": question,
+    }
+
+    result = await runtime.run("question", params, inference_options)
+    print(f"result:\n{result}")
+    return result
+
+
+async def get_mod_result(source_code: str, question: str):
+    question_about_code = (
+        f"QUERY ABOUT SOURCE CODE:\n{question}\nSOURCE CODE:\n```{source_code}\n```"
+    )
+
+    return await query(question_about_code)
+
+
+async def mod_code(source_code_file: str, question: str, update_file: bool = False):
+    # read source code from file
+    with open(source_code_file, "r") as file:
+        source_code = file.read()
+
+        answer = await get_mod_result(source_code, question)
+        s_answer = "the anser"
+
+        if update_file:
+            # Here you would add your logic related to how the original code needs to be modified based on the answer
+            with open(source_code_file, "w") as file:
+                file.write(s_answer)
+
+    return 0
+
+
+def deprefix(s: str, pfx: str) -> str:
+    if s.startswith(pfx):  # Checks if the string starts with the given prefix
+        return s[len(pfx) :]  # If true, returns the string without the prefix
+    else:
+        return s  # If false, returns the string as is
+
+
+async def mod_loop(source_code_file: str):
+    config_file_path = "cli-mate.aiconfig.json"
+    runtime = AIConfigRuntime.load(config_file_path)
+
+    current_task = asyncio.current_task()
+
+    def handler():
+        # print(f"{signum} received")
+        print(f"SIGINT received {current_task}")
+        if current_task:
+            current_task.cancel()
+
+    loop = asyncio.get_event_loop()
+    loop.add_signal_handler(signal.SIGINT, handler)
+
+    i = 0
+    while True:
+        try:
+            user_input = input("\nQuery: ")
+            if user_input.strip() == "":
+                continue
+            if user_input.strip().startswith("reload") or i == 0:
+                user_input = deprefix(user_input.strip(), "reload")
+                source_code = open(source_code_file, "r").read()
+                prompt = (
```

Comment:
I think this should be in the aiconfig itself.

See the prompt routing cookbook to see how to route to the right prompt using aiconfig

Reply:
I'm not sure how to do this in a loop - every prompt in this script is added dynamically. Any advice? cc @LastMileNewGrad